src/components/device/ha-device-trigger-picker.ts

```diff
@@ -0,0 +1,185 @@
+import "@polymer/paper-input/paper-input";
+import "@polymer/paper-item/paper-item";
+import "@polymer/paper-item/paper-item-body";
+import "@polymer/paper-dropdown-menu/paper-dropdown-menu-light";
+import "@polymer/paper-listbox/paper-listbox";
+import {
+  LitElement,
+  TemplateResult,
+  html,
+  css,
+  CSSResult,
+  customElement,
+  property,
+} from "lit-element";
+import { HomeAssistant } from "../../types";
+import { fireEvent } from "../../common/dom/fire_event";
+import computeStateName from "../../common/entity/compute_state_name";
+import {
+  DeviceTrigger,
+  fetchDeviceTriggers,
+  triggersEqual,
+} from "../../data/device_automation";
+
+@customElement("ha-device-trigger-picker")
+class HaDeviceTriggerPicker extends LitElement {
+  public hass?: HomeAssistant;
+  @property() public label?: string;
+  @property() public deviceId?: string;
+  @property() public triggers: { [key: string]: DeviceTrigger } = {};
+  public presetTrigger?: DeviceTrigger;
+
+  private noTrigger: DeviceTrigger = {
+    device_id: "",
+    platform: "device",
+    domain: "",
+    entity_id: "",
+  };
+  private unknownTrigger?: DeviceTrigger = {
+    device_id: "",
+    platform: "device",
+    domain: "",
+    entity_id: "",
+  };
+  private key?: string;
+  private setTrigger?: DeviceTrigger;
+
+  private readonly NO_TRIGGER = "NO_TRIGGER";
+  private readonly UNKNOWN_TRIGGER = "UNKNOWN_TRIGGER";
+
+  protected render(): TemplateResult | void {
+    const noTriggers = Object.keys(this.triggers).length === 0;
+    return html`
+      <paper-dropdown-menu-light .label=${this.label} ?disabled=${noTriggers}>
+        <paper-listbox
+          slot="dropdown-content"
+          .selected=${this.key}
+          attr-for-selected="key"
+          @iron-select=${this._triggerChanged}
+          id="listbox"
+        >
+          <paper-item .key=${this.NO_TRIGGER} .trigger=${this.noTrigger} hidden>
+            No triggers
+          </paper-item>
+          <paper-item
+            .key=${this.UNKNOWN_TRIGGER}
+            .trigger=${this.unknownTrigger}
+            hidden
```

Comment:
Why is this one hidden ?

Reply:
because it should be shown in the dropdown, but not be possible to select.

Reply:
Ah check !